ldelf/ldelf.mk

```diff
@@ -7,7 +7,10 @@ sm-$(sm) := y
 link-out-dir$(sm) := $(out-dir)/$(sm)
 
 cppflags$(sm)	:= $(core-platform-cppflags)
-cflags$(sm)	:= $(core-platform-cflags) -fpie -fvisibility=hidden
+cflags$(sm)	:= $(core-platform-cflags) -fvisibility=hidden
+ifeq ($(CFG_TA_ASLR),y)
+cflags$(sm) += -pie
```

Comment:
`-fpie` is a linked flag but set in cflags and Clang will complain not using it.
Since `CFG_TA_ASLR` is enabled on `mk/config.mk` I would say there is no need to change it to n for arm. For any other architecture not supproting ASLR not sure about the best way to get rid of this linker flag ensuring compatibility with all existing toolchains.

Reply:
@jforissier : Thank you for correcting the typo error.
Do you think that we still coud have this check to move forward on porting OP-TEE for platforms not having ASLR ? For ARM it will be always enabled anyway.
what you suggested could be arch/platform specific code ?